src/nvim/api/vim.c

```diff
@@ -2354,3 +2356,72 @@ Array nvim__inspect_cell(Integer grid, Integer row, Integer col, Error *err)
   }
   return ret;
 }
+
+/// Invokes the nvim server to read from stdin when it is not a tty
+///
+/// It enables functionalities like:
+/// - echo "1f u c4n r34d th1s u r34lly n33d t0 g37 r357"| nvim -
+/// - cat path/to/a/file | nvim -
+/// It has to be called before |nvim_ui_attach()| is called in order
+/// to ensure proper functioning.
+///
+/// @param channel_id: The channel id of the GUI-client
+/// @param filedesc: The file descriptor of the GUI-client process' stdin
+/// @param[out] err Error details, if any
+/// @return Boolean : true if filedesc was not a tty
+///                   false otherwise 
+Boolean nvim_read_stdin(uint64_t channel_id, Integer filedesc, Error *err)
+FUNC_API_SINCE(6) FUNC_API_REMOTE_ONLY
+{
+  if (remote_ui_get(channel_id)) {
+    // If nvim_ui_attach is already called then abort
+    api_set_error(err, kErrorTypeValidation,
+                  "nvim_ui_attach has already been called");
+    return false;
+  }
+  stdin_isatty = os_isatty((int)filedesc);
+  if (stdin_isatty) {
+    // If the passed filedesc is a tty
+    api_set_error(err, kErrorTypeValidation,
+                  "The passed file descriptor was a tty");
+    return false;
+  }
+  stdin_filedesc = (int)filedesc;
+  return true;
+}
+
+/// Sets if the stdin and stdout are ttyin or not. Used when `nvim` is
+/// launched as a `embed` `job` with its stdin and stout as pipes.
+///
+/// @param channel_id: The channel id of the GUI-client
+/// @param values: A dictionary containing "isatty" info on stdin and
+///                stout of the parent process.
+/// @param[out] err Error details, if any
+/// @return Boolean : false if errored and
+///                   true otherwise
+Boolean nvim_set_stdin_stdout(uint64_t channel_id, Dictionary values, Error *err)
```

Comment:
can't these just be `term_ttyin` and `term_ttyout` UI options?

Reply:
Its a very specific case (has to be `embed` and must be started as a job), hence I hesitated to add it as an ui option. Moreover its specific to `server` and not `UI`.

Reply:
> Its a very specific case

Which is precisely why it should be a `term_` UI option, which by design are specific for when running the server with TUI client. OTOH `nvim_set_stdin_stdout` sounds like it is general functionality.

>  Moreover its an argument for server and not UI.

It is part of the interface between the server and UI. Saying that it "for server and not UI" doesn't mean anything.